Edited the KinematicBody and KinematicBody2D docs

theoway opened the pull request:
move_and_slide method in KinematicBody and KinematicBody2D should be used in _physics_process, for the current engine version, but it is not mentioned in docs. In this pull request, I've edited the docs accordingly.

*Bugsquad edit:* Fixes #36805

Calinou commented on doc/classes/KinematicBody.xml:
```diff
@@ -105,7 +105,8 @@
 			<description>
 				Moves the body along a vector. If the body collides with another, it will slide along the other body rather than stop immediately. If the other body is a [KinematicBody] or [RigidBody], it will also be affected by the motion of the other body. You can use this to make moving or rotating platforms, or to make nodes push other nodes.
-				[code]linear_velocity[/code] is the velocity vector (typically meters per second). Unlike in [method move_and_collide], you should [i]not[/i] multiply it by [code]delta[/code] — the physics engine handles applying the velocity.
+				This method should be used in [method _physics_process].
```
It would be good to explain *why* this is needed, and what happens if you don't:
```suggestion
				This method should be used in [method _physics_process] (or a method called by [method _physics_process]), as it uses the physics step's `delta` value automatically. Otherwise, the simulation won't run at the correct speed.
```

Calinou commented on doc/classes/KinematicBody2D.xml:
```diff
@@ -103,7 +103,8 @@
 			<description>
 				Moves the body along a vector. If the body collides with another, it will slide along the other body rather than stop immediately. If the other body is a [KinematicBody2D] or [RigidBody2D], it will also be affected by the motion of the other body. You can use this to make moving or rotating platforms, or to make nodes push other nodes.
-				[code]linear_velocity[/code] is the velocity vector in pixels per second. Unlike in [method move_and_collide], you should [i]not[/i] multiply it by [code]delta[/code] — the physics engine handles applying the velocity.
+				This method should be used in [method _physics_process].
```
Same as above:

```suggestion
				This method should be used in [method _physics_process] (or a method called by [method _physics_process]), as it uses the physics step's `delta` value automatically. Otherwise, the simulation won't run at the correct speed.
```

theoway replied:
I'll make it more clear. Also correct the method reference.

Calinou commented on doc/classes/KinematicBody.xml:
```diff
@@ -105,7 +105,8 @@
 			<description>
 				Moves the body along a vector. If the body collides with another, it will slide along the other body rather than stop immediately. If the other body is a [KinematicBody] or [RigidBody], it will also be affected by the motion of the other body. You can use this to make moving or rotating platforms, or to make nodes push other nodes.
-				[code]linear_velocity[/code] is the velocity vector (typically meters per second). Unlike in [method move_and_collide], you should [i]not[/i] multiply it by [code]delta[/code] — the physics engine handles applying the velocity.
+				This method should be used in [method Node._physics_process](or in a method called by [method Node._physics_process]), as it uses physics step's 'delta' value automatically in calculations. Otherwise, simulation will run at incorrect speed.
```
```suggestion
				This method should be used in [method Node._physics_process] (or in a method called by [method Node._physics_process]), as it uses the physics step's [code]delta[/code] value automatically in calculations. Otherwise, the simulation will run at an incorrect speed.
```

Calinou commented on doc/classes/KinematicBody2D.xml:
```diff
@@ -103,7 +103,8 @@
 			<description>
 				Moves the body along a vector. If the body collides with another, it will slide along the other body rather than stop immediately. If the other body is a [KinematicBody2D] or [RigidBody2D], it will also be affected by the motion of the other body. You can use this to make moving or rotating platforms, or to make nodes push other nodes.
-				[code]linear_velocity[/code] is the velocity vector in pixels per second. Unlike in [method move_and_collide], you should [i]not[/i] multiply it by [code]delta[/code] — the physics engine handles applying the velocity.
+				This method should be used in [method Node._physics_process](or in a method called by [method Node._physics_process]), as it uses physics step's 'delta' value automatically in calculations. Otherwise, simulation will run at incorrect speed.
```
```suggestion
				This method should be used in [method Node._physics_process] (or in a method called by [method Node._physics_process]), as it uses the physics step's [code]delta[/code] value automatically in calculations. Otherwise, the simulation will run at an incorrect speed.
```

theoway replied:
I've made the changes.

theoway committed:
Edited the KinematicBody and KinematicBody2D docs
Changes made as requested.

Explained the reason why move_and_slide should be sued in _physics_process
Final Changes

Edited the docs as per the changes requested.
Edits in docs

Made some changes

akien-mga commented:
Thanks!

akien-mga commented:
Cherry-picked for 3.2.1.
